src/platform/nrfconnect/FactoryDataProvider.cpp

```diff
@@ -42,6 +43,14 @@ CHIP_ERROR FactoryDataProvider<FlashFactoryData>::Init()
     uint8_t * factoryData = nullptr;
     size_t factoryDataSize;
 
+    // Add write protection to the Factory Data flash area
+    constexpr static uint32_t partitionManagerAreaIsWriteProtected = 1;
+    int ret = fprotect_area(PM_FACTORY_DATA_ADDRESS, PM_FACTORY_DATA_SIZE);
+    if(0 != ret || (partitionManagerAreaIsWriteProtected != fprotect_is_protected(PM_FACTORY_DATA_ADDRESS))){
```

Comment:
Do we need to allocate `partitionManagerAreaIsWriteProtected` instead of just `1 == `?

Reply:
In fact for compiler it will be the same, as both are just 4B consts, so maybe it's not a big deal. I was just surprised why you do that, but I assume it's just to improve code readability.

Reply:
Yes, it's only for code readability.